services/121-service/src/payments/payments.service.spec.ts

```diff
@@ -0,0 +1,128 @@
+import { ProgramEntity } from '@121-service/src/programs/program.entity';
+import { RegistrationStatusEnum } from '@121-service/src/registration/enum/registration-status.enum';
+import { RegistrationScopedRepository } from '@121-service/src/registration/repositories/registration-scoped.repository';
+
+function createMockTransaction(
+  id: number,
+  amount: number,
+  status: TransactionStatusEnum,
+) {
+  return {
+    id,
+    registrationId: id,
+    registrationStatus: RegistrationStatusEnum.included,
+    amount,
+    status,
+    created: new Date(),
+    updated: new Date(),
+    payment: 1,
+    registrationProgramId: id + 100,
+    referenceId: `REF-${id}`,
+    errorMessage: null,
+    programFinancialServiceProviderConfigurationName: 'someFsp',
+  };
+}
+
+const mockTransactions = [
+  createMockTransaction(101, 100, TransactionStatusEnum.success),
+  createMockTransaction(102, 200, TransactionStatusEnum.success),
+];
+
+describe('PaymentsService - getTransactions', () => {
+  let service: PaymentsService;
+  let transactionScopedRepository: TransactionScopedRepository;
+  let programRepository: Repository<ProgramEntity>;
+  let registrationScopedRepository: RegistrationScopedRepository;
+
+  const mockFullnameNamingConvention = ['firstName', 'lastName'];
+
+  beforeEach(async () => {
+    const { unit, unitRef } = TestBed.create(PaymentsService).compile();
+
+    transactionScopedRepository = unitRef.get(TransactionScopedRepository);
+    registrationScopedRepository = unitRef.get(RegistrationScopedRepository);
+    programRepository = unitRef.get('ProgramEntityRepository');
+    service = unit;
+
+    jest.spyOn(transactionScopedRepository, 'getTransactionsForPayment');
+    jest.spyOn(programRepository, 'findOneOrFail');
+    jest.spyOn(registrationScopedRepository, 'getFullNamesByRegistrationIds');
+  });
+
+  it('should return transactions with names when fullnameNamingConvention exists', async () => {
+    // Arrange
+    const programId = 1;
+    const payment = 2;
+
+    const mockRegistrationNames = [
+      { registrationId: 101, name: 'John Doe' },
+      { registrationId: 102, name: 'Jane Smith' },
+    ];
+
+    jest
+      .spyOn(transactionScopedRepository, 'getTransactionsForPayment')
+      .mockResolvedValue(mockTransactions);
+    jest.spyOn(programRepository, 'findOneOrFail').mockResolvedValue({
+      fullnameNamingConvention: mockFullnameNamingConvention,
+    } as ProgramEntity);
+    jest
+      .spyOn(registrationScopedRepository, 'getFullNamesByRegistrationIds')
+      .mockResolvedValue(mockRegistrationNames);
+
+    // Act
+    const result = await service.getTransactions({ programId, payment });
+
+    // Assert
+    expect(result).toEqual([
+      { ...mockTransactions[0], name: 'John Doe' },
+      { ...mockTransactions[1], name: 'Jane Smith' },
+    ]);
+  });
+
+  it('should return transactions without names when fullnameNamingConvention is empty', async () => {
```

Comment:
Does this happen in production projects?

Reply:
No, but it is still technically allowed to have multiple names. I think we should still cover it with tests, unless we remove/refactor the fullnameNamingConvention